packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts

```diff
@@ -370,6 +430,52 @@ export class NgOptimizedImage implements OnInit, OnChanges, OnDestroy {
     return finalSrcs.join(', ');
   }
 
+  private getAutomaticSrcset(): string {
+    if (this.sizes) {
+      return this.getResponsiveSrcset();
+    } else {
+      return this.getFixedSrcset();
+    }
+  }
+
+  private getResponsiveSrcset(): string {
+    const {viewportBreakpoints, subViewportBreakpoints} = this.config;
+
+    let breakpoints: number[];
+    if (this.sizes?.trim() === '100vw') {
+      // Since this is a full-screen-width image, our srcset only needs to include
+      // breakpoints with full viewport widths.
+      breakpoints = viewportBreakpoints!;
+    } else {
+      breakpoints = subViewportBreakpoints!.concat(viewportBreakpoints!);
+      // Trim off any value too small to be relevant. This is determined by finding
+      // the smallest percent size in the `sizes` prop, multiplying it by the smallest
+      // size in the viewport prop, and removing anything smaller from the subviewport
+      // breakpoints.
+      const vwRegExp = /(^|\s)(1?\d?\d)vw/g;
+      let percentSizeMatch;
+      const sizeRatios = [];
+      while (percentSizeMatch = vwRegExp.exec(this.sizes!)) {
+        sizeRatios.push(parseInt(percentSizeMatch[2]) * 0.01);
```

Comment:
I think we can avoid an intermediate array here without affecting the reasability:
```suggestion
      let smallestRatio: number|null = null;
      let percentSizeMatch;
      while (percentSizeMatch = vwRegExp.exec(this.sizes!)) {
        smallestRatio = Math.min(smallestRatio, parseInt(percentSizeMatch[2]) * 0.01);
      }
      if (smallestRatio !== null) { ... }
```

Reply:
Looks good! Updated